plugins/common/shim/output.go

```diff
@@ -3,12 +3,43 @@ package shim
 import (
 	"bufio"
 	"fmt"
+	"os"
+	"sync"
+	"time"
 
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/models"
 	"github.com/influxdata/telegraf/plugins/parsers/influx"
 )
 
+type batchMetrics struct {
```

Comment:
This should better be in its own file together with specific tests for each of the methods (`add`, `clear`, `len`)